core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt

```diff
@@ -40,10 +46,45 @@ internal fun <T, C> Update<T, C>.updateWithValuePerColumnImpl(selector: Selector
         }
     }
 
+/**
+ * Implementation for Update As Frame:
+ * Replaces selected column groups with the result of the expression only where the filter is true.
+ */
+internal fun <T, C, R> Update<T, DataRow<C>>.asFrameImpl(expression: DataFrameExpression<C, DataFrame<R>>): DataFrame<T> =
+    if (df.isEmpty()) df
+    else df.replace(columns).with {
+        val src = it.asColumnGroup()
+        val updatedColumn = expression(src, src).asColumnGroup(src.name())
+        if (filter == null) {
+            // If there is no filter, we simply replace the selected column groups with the result of the expression
+            updatedColumn
+        } else {
+            // If there is a filter, we collect the indices of the rows that are inside the filter
+            val collector = createDataCollector<DataRow<C>>(it.size, it.type)
+            val indices = buildList {
+                df.indices().forEach { rowIndex ->
+                    val row = AddDataRowImpl(rowIndex, df, collector.values)
+                    val currentValue = row[src]
+
+                    if (filter.invoke(row, currentValue)) {
+                        this += rowIndex
+                        collector.add(currentValue)
+                    }
+                }
+            }
+
+            // Then we only replace the original rows with the updated rows that are inside the filter
+            src.rows().map {
```

Comment:
could we extract it to the separate function with the name related to this comment (to be sure that it mutates the data in the dataframe from the function signature, for example)

Reply:
sure